runtime/doc/lsp.txt

```diff
@@ -0,0 +1,265 @@
+*lsp.txt* The Language Server Protocol
+
+        NVIM REFERENCE MANUAL
+
+
+Neovim Language Server Protocol (LSP) API
+
+Neovim exposes a powerful API that conforms to Microsoft's published Language
+Server Protocol specification. The documentation can be found here:
+
+    https://github.com/Microsoft/language-server-protocol
+
+================================================================================
+                                                                  *lsp-request*
+Requests
+
+To make a request of the language server client running in Nvim, the following
+API is defined:
+>
+    :call lsp#request(method, arguments, callback, filetype)
+    :call lsp#request_async(method, arguments, callback, filetype)
+<
+
+                                                                *lsp#request()*
+lsp#request({method} [, {arguments} [, {callback} [, {filetype}]]])
```

Comment:
the existence of `lsp#request_async` seems to imply that this is synchronous. So the response data is _both_ passed to a callback _and_ returned by this function? That seems a bit confusing.

Reply:
callback is not working as lua requires a function pointer lsp#request_async send string